bolt-design-system/bolt
# Ratio Component Refactor + Jest Testing Coverage

## Jira
http://vjira2:8080/browse/BDS-987

## Summary
- Refreshes the ratio component's internals + adds testing coverage support via Jest (including visual regression testing)


![image](https://user-images.githubusercontent.com/1617209/53360052-8d02ce80-3902-11e9-8b91-dc268559abbf.png)
![ratio-js-bolt-ratio-component-bolt-ratio-with-html-5-video-renders-1-snap](https://user-images.githubusercontent.com/1617209/53360061-93914600-3902-11e9-8e37-f37dcd1d8e46.png)
> Screenshot snapshots for VRT in the `<bolt-ratio>` component

## Details
- Cleans up and refactors the ratio component to simplify rendering logic + allow for more easily testing the Shadow DOM-related rendering behavior + conditional behavior based on support for CSS Custom properties  
- Adds several new Jest tests to the Ratio component — checking the ratio component’s Twig rendering and the web component rendering!
- Also adds two new visual regression tests to the ratio component (one with a vanilla HTML5 `<video>` example and the other with a vanilla `<img>` tag example)
- Updates then `.boltrc` config + Jest-related logic to copy over the example video asset used for demos & testing
- Updates the Jest test fixture logic to auto-copy over any non-image assets (previously these were not copied over)
  - Fixes an issue where the Jest testing environment didn’t have the Bolt webfonts correctly copied over + wasn’t allowed to access these assets when being served from the `www` folder
  - This should also address previous VRT tests / test differences noticed with the button component
- Tests out adding a few component-specific examples to the component's schema file directly (potential emerging pattern on how we can handle demoing some of these examples automatically)
- Updates the `<bolt-ratio>` component schema to differentiate between Web Component vs Twig-specific props
- Adjusted the docs site schema table logic to better differentiate between the deprecated vs non-deprecated prop options

## Deprecated
The ratio component's `aspect-ratio-height` and `aspect-ratio-width` props are deprecated and will be removed in Bolt v3.0.

## How to test
- Confirm all Travis CI build tests passing
- Spot check other component examples and page prototypes to verify consistent behavior / styling
